parquet/src/main/java/org/apache/iceberg/parquet/ReadConf.java

```diff
@@ -165,12 +165,12 @@ ParquetFileReader reader() {
     return shouldSkip;
   }
 
-  private Map<Long, Long> generateOffsetToStartPos() {
-    ParquetFileReader fileReader = newReader(this.file, ParquetReadOptions.builder().build());
```

Comment:
@sudssf This needs to initialize a new reader as we want to get row positions relative to the start of the file and not start of the split. I think the only change that is necessary in this PR is to close the reader.

Reply:
ok I will revert to previous commit thanks for reply

Reply:
Can you remove the extra newline here?